aries_cloudagent/registrar/default/indy.py

```diff
@@ -138,69 +136,95 @@ async def create(
                     verkey,
                     alias,
                     role,
-                    write_ledger=write_ledger,
+                    write_ledger= endorsed, # endorsers will write txn and not authors
                     endorser_did=endorser_did,
                 )
             except LedgerTransactionError as err:
-                raise RegistrarError(err.roll_up)
+                raise RegistrarError(err.roll_up) from err
             except LedgerError as err:
-                raise RegistrarError(err.roll_up)
+                raise RegistrarError(err.roll_up) from err
             except WalletNotFoundError as err:
-                raise RegistrarError(err.roll_up)
+                raise RegistrarError(err.roll_up) from err
             except WalletError as err:
                 raise RegistrarError(
                     (
                         f"Registered NYM for DID {did} on ledger but could not "
                         f"replace metadata in wallet: {err.roll_up}"
                     )
-                )
+                ) from err
+        return (success, txn)
 
-        meta_data = {"verkey": verkey, "alias": alias, "role": role}
-        if not create_transaction_for_endorser:
-            # Notify event
-            await notify_did_event(profile, did, meta_data)
-            # return web.json_response({"success": success})
 
-            # TODO Determine what the actual state is here
-            return JobRecord(state=JobRecord.STATE_REGISTERED)
-        else:
-            transaction_mgr = TransactionManager(profile)
-            try:
-                transaction = await transaction_mgr.create_record(
-                    messages_attach=txn["signed_txn"],
-                    connection_id=connection_id,
-                    meta_data=meta_data,
+    async def create(
+        self,
+        profile: Profile,
+        method: Optional[str],
+        did: Optional[str] = None,
+        document: Optional[dict] = None,
+        **options: dict,
+    ) -> JobRecord:
+        """Create a DID from a given method."""
+        raise NotImplementedError
```

Comment:
Errors should be instances, i.e.:
```suggestion
        raise NotImplementedError()
```